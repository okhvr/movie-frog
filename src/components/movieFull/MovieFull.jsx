import React, { Component } from 'react';

import './style.scss';

export default class MovieFull extends Component {
    render() {
        const {movie} = this.props;
        return (
            <div className="card mb-3">
                <div className="row no-gutters">
                    <div className="col-md-4">
                        <img src={movie.poster_path} className="card-img" />
                    </div>
                    <div className="col-md-8">
                        <div className="card-body">
                            <h5 className="card-title">{movie.title}</h5>
                            <h5 className="card-title">
                                <span className="badge badge-secondary">{movie.release_date}</span>
                            </h5>
                            <p className="card-text">{movie.overview}</p>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}