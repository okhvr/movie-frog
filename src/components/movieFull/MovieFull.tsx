import { Component } from 'react';
import * as React from 'react';

import { MovieType } from '../../types';

type MyProps = {
    movie: MovieType,
};

export default class MovieFull extends Component {
    props: MyProps;

    render() {
        const {movie} = this.props;
        return (
            <div className="card mb-3 bg-transparent">
                <div className="row no-gutters">
                    <div className="col-md-4">
                        <img src={movie.poster_path} className="card-img" />
                    </div>
                    <div className="col-md-8">
                        <div className="card-body">
                            <h5 className="card-title">{movie.title}</h5>
                            <h5 className="card-title">
                                <span className="badge badge-secondary">{movie.release_date}</span>
                            </h5>
                            <p className="card-text">{movie.overview}</p>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}