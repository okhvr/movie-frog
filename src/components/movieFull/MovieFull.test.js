import React from 'react';
import renderer from 'react-test-renderer';
import MovieFull from './MovieFull';

describe('MovieFull component', () => {
    const movie = {
        id: '1',
        poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
        title: 'test',
        release_date: '12/06/1997',
        overview: 'testtest'
    };
    it('should match with snapshot', () => {
        const component = renderer.create(
            <MovieFull movie={movie}/>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});