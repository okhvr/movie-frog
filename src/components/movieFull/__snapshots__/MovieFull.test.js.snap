// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieFull component should match with snapshot 1`] = `
<div
  className="card mb-3 bg-transparent"
>
  <div
    className="row no-gutters"
  >
    <div
      className="col-md-4"
    >
      <img
        className="card-img"
        src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
      />
    </div>
    <div
      className="col-md-8"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          test
        </h5>
        <h5
          className="card-title"
        >
          <span
            className="badge badge-secondary"
          >
            12/06/1997
          </span>
        </h5>
        <p
          className="card-text"
        >
          testtest
        </p>
      </div>
    </div>
  </div>
</div>
`;
